sfa_dash/blueprints/reports.py

```diff
@@ -125,36 +121,113 @@ def post(self):
-        return {'report': report_template,
-                'includes_bokeh': True}
+        include_timeseries = self.should_include_timeseries()
+        report_template, report_kwargs = get_template_and_kwargs(
+            self.metadata,
+            request.url_root.rstrip('/'),
+            include_timeseries,
+            True
+        )
+        report_kwargs.update({
+            'report_template': report_template,
+            'dash_url': request.url_root.rstrip('/'),
+            'includes_bokeh': True
+        })
+        if not include_timeseries:
+            # display a message about omitting timeseries
+            download_link = url_for('data_dashboard.download_report_html',
+                                    uuid=self.metadata['report_id'])
+            script = ''
+            div = f"""<div class="alert alert-warning">
+    <strong>Warning</strong> To improve performance timeseries plots have been
+    omitted from this report. You may download a copy of this report with the
+    timeseries plots included:
+    <a href="{download_link}">Download HTML Report.</a></div>"""
+            report_kwargs.update({
+                'timeseries_div': div,
+                'timeseries_script': script,
+            })
+        return report_kwargs
 
-    def get(self, uuid):
+    def set_metadata(self, uuid):
+        """Loads all necessary data for loading a
+        `solarforecastarbiter.datamodel.Report` with processed forecasts and
+        observations.
+        """
+        metadata = handle_response(reports.get_metadata(uuid))
+        pairs = []
+        for fxobs in metadata['report_parameters']['object_pairs']:
+            pair = {}
+            if 'observation' in fxobs:
+                observation = handle_response(
+                    observations.get_metadata(fxobs['observation']))
+                observation['site'] = handle_response(
+                    sites.get_metadata(observation['site_id']))
+                pair['observation'] = observation
+            if 'aggregate' in fxobs:
+                aggregate = handle_response(
+                    aggregates.get_metadata(fxobs['aggregate']))
+                obs_list = []
+                for obs in aggregate['observations']:
+                    observation = handle_response(
+                        observations.get_metadata(obs))
+                    observation['site'] = handle_response(
+                        sites.get_metadata(observation['site_id']))
+                    obs_list.append(observation)
+                aggregate['observations'] = obs_list
+                pair['aggregate'] = aggregate
+            forecast = handle_response(
+                forecasts.get_metadata(fxobs['forecast']))
+            forecast['site'] = handle_response(
+                sites.get_metadata(forecast['site_id']))
+            pair['forecast'] = forecast
+            pairs.append(pair)
+        metadata['report_parameters']['object_pairs'] = pairs
```

Comment:
getting the actual object pairs is unnecessary to render the report. You can just remove this loop and set object pairs to an empty list